jms/src/main/scala/akka/stream/alpakka/jms/JmsConnector.scala

```diff
@@ -60,20 +65,18 @@ private[jms] trait JmsConnector[S <: JmsSession] {
   case object Connected extends ConnectionStatus
   case object TimedOut extends ConnectionStatus
 
-  protected def initSessionAsync(withReconnect: Boolean = true): Unit = {
-
-    def failureHandler(ex: Throwable) =
-      if (withReconnect && ex.isInstanceOf[jms.JMSException]) initSessionAsync()
-      else fail.invoke(ex)
+  protected def initSessionAsync(): Unit = {
 
-    val allSessions = openSessions(failureHandler)
-    allSessions.failed.foreach(failureHandler)
+    val allSessions = openSessions()
+    allSessions.failed.foreach(connectionFailedCB.invoke)
```

Comment:
here, I think we must switch back to the graph stage threads with the asyncCallback